go/arrow/array/string.go

```diff
@@ -50,8 +50,11 @@ func (a *String) Reset(data arrow.ArrayData) {
 
 // Value returns the slice at index i. This value should not be mutated.
 func (a *String) Value(i int) string {
+	sb := strings.Builder{}
+
 	i = i + a.array.data.offset
-	return a.values[a.offsets[i]:a.offsets[i+1]]
+	sb.WriteString(a.values[a.offsets[i]:a.offsets[i+1]])
+	return sb.String()
```

Comment:
There's no need to use the string builder here, you can just do `string(a.values[a.offsets[i]:a.offsets[i+1]])`